modules/file_upload/cms_file_uploader_module.e

```diff
@@ -0,0 +1,332 @@
+note
+	description: "file_upload application root class"
+	date: "$Date$"
+	revision: "$Revision$"
+
+class
+	CMS_FILE_UPLOADER_MODULE
+
+inherit
+	CMS_MODULE
+		rename
+			module_api as file_upload_api
+		redefine
+			install,
+			initialize,
+			setup_hooks,
+			permissions,
+			file_upload_api
+		end
+
+	CMS_HOOK_MENU_SYSTEM_ALTER
+
+	SHARED_EXECUTION_ENVIRONMENT
+
+create
+	make
+
+feature {NONE} -- Initialization
+
+	make
+		do
+			name := "file_uploader"
+			version := "1.0"
+			description := "Service to upload files, and manage them."
+			package := "file"
+		end
+
+feature -- Access
+
+	name: STRING
+
+	permissions: LIST [READABLE_STRING_8]
+			-- List of permission ids, used by this module, and declared.
+		do
+			Result := Precursor
+			Result.force ("admin uploaded files")
```

Comment:
I removed the previous permissions which were not related to this module.

And now, the permissions should also be checked in the related code, which is not the case for now !